astropy/time/core.py

```diff
@@ -1180,6 +1180,36 @@ def __setitem__(self, item, value):
         self._time.jd1[item] = value._time.jd1
         self._time.jd2[item] = value._time.jd2
 
+    def isclose(self, other, atol=None):
+        """Returns a boolean or boolean array where two Time objects are
+        element-wise equal within a time tolerance.
+
+        This evaluates the expression below::
+
+          abs(self - other) <= atol
+
+        Parameters
+        ----------
+        other : `~astropy.time.Time`
+            Time object for comparison.
+        atol : `~astropy.units.Quantity` or `~astropy.time.TimeDelta`
+            Absoute tolerance for equality with units of time (e.g. ``u.s`` or
+            ``u.day``). Default is one bit in the 128-bit JD time representation,
```

Comment:
I wonder a bit about the default: when we were doing the hypothesis testing, I found that for UTC I had to allow 2 bits (since taking the difference with a UTC time means one has to do a scale change): 
https://github.com/astropy/astropy/blob/f56d8abf5dd39671c695e1e295293ef1490e2485/astropy/time/tests/test_precision.py#L502-L512

Maybe we should do that here too? (Also, I guess in principle, one can just put the default in in the initializer, rather than make it `None`.)

Reply:
I made the change.

About putting the atol in the signature, I thought about this and am on the edge, but somehow seeing a somewhat non-trivial expression in the function signature feels wrong: 
```
def isclose(self, other, atol=2 * np.finfo(float).eps * u.day):
```
I think the repr of `<Quantity 4.4408921e-16 d>` won't really be that helpful in the method docstring.

Reply:
I'd definitely leave it in `ns` if we put it in the signature... But, yes, clearly a user would have to read the docstring anyway, so really it doesn't matter - let's stick with what you have.